google/cloud/storage/_helpers.py

```diff
@@ -297,3 +304,52 @@ def _base64_md5hash(buffer_object):
     _write_buffer_to_hash(buffer_object, hash_obj)
     digest_bytes = hash_obj.digest()
     return base64.b64encode(digest_bytes)
+
+
+def _base64_crc32chash(buffer_object):
+    """Get crc32c hash of bytes (as base64).
+
+    :type buffer_object: bytes buffer
+    :param buffer_object: Buffer containing bytes used to compute an MD5
+                          hash (as base64).
+
+    :rtype: str
+    :returns: A base64 encoded digest of the MD5 hash.
+    """
+    hash_obj = _get_crc32c_object()
+    _write_buffer_to_hash(buffer_object, hash_obj)
```

Comment:
This asserts that the object is being hashed before upload right?

Reply:
I believe that is where this is ultimately used yes. This is near identical to the calculation to the md5 hashing above as the two hashes provide a shared interface.